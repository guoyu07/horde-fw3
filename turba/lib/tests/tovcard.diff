003- FN;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:Jan Schneider=F6
003+ FN;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:=
004- EMAIL:jan@horde.org
004+ Jan Schneider=F6
005- NICKNAME:yunosh
005+ EMAIL:jan@horde.org
006- LABEL;HOME;ENCODING=QUOTED-PRINTABLE;CHARSET=ISO-8859-1:Sch=F6nestr. 15=0D=0A=
006+ NICKNAME:yunosh
007- 33604 Bielefeld
007+ LABEL;HOME;ENCODING=QUOTED-PRINTABLE;CHARSET=ISO-8859-1:=
008- TEL;HOME:+49 521 555123
008+ Sch=F6nestr. 15=0D=0A33604 Bielefeld
009- TEL;WORK:+49 521 555456
009+ TEL;HOME:+49 521 555123
010- TEL;CELL:+49 177 555123
010+ TEL;WORK:+49 521 555456
011- TEL;FAX:+49 521 555789
011+ TEL;CELL:+49 177 555123
012- TEL;PAGER:+49 123 555789
012+ TEL;FAX:+49 521 555789
013- BDAY:1971-10-01
013+ TEL;PAGER:+49 123 555789
014- TITLE;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:Senior Developer (=E4=F6=FC)
014+ BDAY:1971-10-01
015- ROLE;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:Developer (=E4=F6=FC)
015+ TITLE;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:=
016- NOTE;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:A German guy (=E4=F6=FC)
016+ Senior Developer (=E4=F6=FC)
017- URL:http://janschneider.de
017+ ROLE;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:=
018- TZ;VALUE=text:Europe/Berlin
018+ Developer (=E4=F6=FC)
019- GEO:13.377778,52.516276
019+ NOTE;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:=
020- PHOTO;ENCODING=b;TYPE=image/png:iVBORw0KGgoAAAANSUhEUgAAAAkAAAAJAgMAAACd/+6DAAAACVBMVEW6ABZmZmYAAACMtcxCAAAAAXRSTlMAQObYZgAAABpJREFUCFtjYACBBgYmRgEIZmGBYAFGMAYBABVmAOEH9qP8AAAAAElFTkSuQmCC
020+ A German guy (=E4=F6=FC)
021- N;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:Schneider=F6;Jan;K.;Mr.;
021+ URL:http://janschneider.de
022- ORG;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:Horde Project;=E4=F6=FC
022+ TZ;VALUE=text:Europe/Berlin
023- ADR;HOME;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:;;Sch=F6nestr. 15=0D=0A=
023+ GEO:52.516276,13.377778
024- 33604 Bielefeld;;;;
024+ PHOTO;ENCODING=b;TYPE=image/png:iVBORw0KGgoAAAANSUhEUgAAAAkA
025- ADR;WORK;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:;;H=FCbschestr. 19;K=F6ln;Allg=E4u;;D=E4nemark
025+  AAAJAgMAAACd/+6DAAAACVBMVEW6ABZmZmYAAACMtcxCAAAAAXRSTlMAQObY
026- END:VCARD
026+  ZgAAABpJREFUCFtjYACBBgYmRgEIZmGBYAFGMAYBABVmAOEH9qP8AAAAAElF
027- 
027+  TkSuQmCC
028- BEGIN:VCARD
028+ N;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:=
029- VERSION:3.0
029+ Schneider=F6;Jan;K.;Mr.;
030- FN:Jan Schneiderö
030+ ORG;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:=
031- EMAIL:jan@horde.org
031+ Horde Project;=E4=F6=FC
032- NICKNAME:yunosh
032+ ADR;HOME;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:=
033- LABEL;TYPE=HOME:Schönestr. 15\n33604 Bielefeld
033+ ;;Sch=F6nestr. 15=0D=0A33604 Bielefeld;;;;
034- TEL;TYPE=HOME:+49 521 555123
034+ ADR;WORK;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:=
035- TEL;TYPE=WORK:+49 521 555456
035+ ;;H=FCbschestr. 19;K=F6ln;Allg=E4u;;D=E4nemark
036- TEL;TYPE=CELL:+49 177 555123
036+ BODY:
037- TEL;TYPE=FAX:+49 521 555789
037+ END:VCARD
038- TEL;TYPE=PAGER:+49 123 555789
038+ 
039- BDAY:1971-10-01
039+ BEGIN:VCARD
040- TITLE:Senior Developer (äöü)
040+ VERSION:3.0
041- ROLE:Developer (äöü)
041+ FN:Jan Schneiderö
042- NOTE:A German guy (äöü)
042+ EMAIL:jan@horde.org
043- URL:http://janschneider.de
043+ NICKNAME:yunosh
044- TZ;VALUE=text:Europe/Berlin
044+ LABEL;TYPE=HOME:Schönestr. 15\n33604 Bielefeld
045- GEO:52.516276;13.377778
045+ TEL;TYPE=HOME:+49 521 555123
046- PHOTO;ENCODING=b;TYPE=image/png:wolQTkcNChoKAAAADUlIRFIAAAAJAAAACQIDAAAAwp3
046+ TEL;TYPE=WORK:+49 521 555456
047-  Dv8OuwoMAAAAJUExURcK6ABZmZmYAAADCjMK1w4xCAAAAAXRSTlMAQMOmw5hmAAAAGklEQVQIW2
047+ TEL;TYPE=CELL:+49 177 555123
048-  NgAMKBBgYmRgEIZmHCgWABRjAGAQAVZgDDoQfDtsKjw7wAAAAASUVORMKuQmDCgg==
048+ TEL;TYPE=FAX:+49 521 555789
049- N:Schneiderö;Jan;K.;Mr.;
049+ TEL;TYPE=PAGER:+49 123 555789
050- ORG:Horde Project;äöü
050+ BDAY:1971-10-01
051- ADR;TYPE=HOME:;;Schönestr. 15\n33604 Bielefeld;;;;
051+ TITLE:Senior Developer (äöü)
052- ADR;TYPE=WORK:;;Hübschestr. 19;Köln;Allgäu;;Dänemark
052+ ROLE:Developer (äöü)
053- END:VCARD
053+ NOTE:A German guy (äöü)
054+ URL:http://janschneider.de
055+ TZ;VALUE=text:Europe/Berlin
056+ GEO:52.516276,13.377778
057+ PHOTO;ENCODING=b;TYPE=image/png:wolQTkcNChoKAAAADUlIRFIAAAAJ
058+  AAAACQIDAAAAwp3Dv8OuwoMAAAAJUExURcK6ABZmZmYAAADCjMK1w4xCAAAA
059+  AXRSTlMAQMOmw5hmAAAAGklEQVQIW2NgAMKBBgYmRgEIZmHCgWABRjAGAQAV
060+  ZgDDoQfDtsKjw7wAAAAASUVORMKuQmDCgg==
061+ N:Schneiderö;Jan;K.;Mr.;
062+ ORG:Horde Project;äöü
063+ ADR;TYPE=HOME:;;Schönestr. 15\n33604 Bielefeld;;;;
064+ ADR;TYPE=WORK:;;Hübschestr. 19;Köln;Allgäu;;Dänemark
065+ BODY:
066+ END:VCARD