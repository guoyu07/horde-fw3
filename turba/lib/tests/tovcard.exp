BEGIN:VCARD
VERSION:2.1
FN;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:Jan Schneider=F6
EMAIL:jan@horde.org
NICKNAME:yunosh
LABEL;HOME;ENCODING=QUOTED-PRINTABLE;CHARSET=ISO-8859-1:Sch=F6nestr. 15=0D=0A=
33604 Bielefeld
TEL;HOME:+49 521 555123
TEL;WORK:+49 521 555456
TEL;CELL:+49 177 555123
TEL;FAX:+49 521 555789
TEL;PAGER:+49 123 555789
BDAY:1971-10-01
TITLE;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:Senior Developer (=E4=F6=FC)
ROLE;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:Developer (=E4=F6=FC)
NOTE;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:A German guy (=E4=F6=FC)
URL:http://janschneider.de
TZ;VALUE=text:Europe/Berlin
GEO:13.377778,52.516276
PHOTO;ENCODING=b;TYPE=image/png:iVBORw0KGgoAAAANSUhEUgAAAAkAAAAJAgMAAACd/+6DAAAACVBMVEW6ABZmZmYAAACMtcxCAAAAAXRSTlMAQObYZgAAABpJREFUCFtjYACBBgYmRgEIZmGBYAFGMAYBABVmAOEH9qP8AAAAAElFTkSuQmCC
N;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:Schneider=F6;Jan;K.;Mr.;
ORG;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:Horde Project;=E4=F6=FC
ADR;HOME;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:;;Sch=F6nestr. 15=0D=0A=
33604 Bielefeld;;;;
ADR;WORK;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:;;H=FCbschestr. 19;K=F6ln;Allg=E4u;;D=E4nemark
END:VCARD

BEGIN:VCARD
VERSION:3.0
FN:Jan Schneiderö
EMAIL:jan@horde.org
NICKNAME:yunosh
LABEL;TYPE=HOME:Schönestr. 15\n33604 Bielefeld
TEL;TYPE=HOME:+49 521 555123
TEL;TYPE=WORK:+49 521 555456
TEL;TYPE=CELL:+49 177 555123
TEL;TYPE=FAX:+49 521 555789
TEL;TYPE=PAGER:+49 123 555789
BDAY:1971-10-01
TITLE:Senior Developer (äöü)
ROLE:Developer (äöü)
NOTE:A German guy (äöü)
URL:http://janschneider.de
TZ;VALUE=text:Europe/Berlin
GEO:52.516276;13.377778
PHOTO;ENCODING=b;TYPE=image/png:wolQTkcNChoKAAAADUlIRFIAAAAJAAAACQIDAAAAwp3
 Dv8OuwoMAAAAJUExURcK6ABZmZmYAAADCjMK1w4xCAAAAAXRSTlMAQMOmw5hmAAAAGklEQVQIW2
 NgAMKBBgYmRgEIZmHCgWABRjAGAQAVZgDDoQfDtsKjw7wAAAAASUVORMKuQmDCgg==
N:Schneiderö;Jan;K.;Mr.;
ORG:Horde Project;äöü
ADR;TYPE=HOME:;;Schönestr. 15\n33604 Bielefeld;;;;
ADR;TYPE=WORK:;;Hübschestr. 19;Köln;Allgäu;;Dänemark
END:VCARD