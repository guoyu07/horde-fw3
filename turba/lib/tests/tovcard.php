<?php

require dirname(__FILE__) . '/../Object.php';
require dirname(__FILE__) . '/../Driver.php';

$attributes = array(
  'name' => 'Jan Schneiderö',
  'namePrefix' => 'Mr.',
  'firstname' => 'Jan',
  'middlenames' => 'K.',
  'lastname' => 'Schneiderö',
  'email' => 'jan@horde.org',
  'alias' => 'yunosh',
  'homeAddress' => 'Schönestr. 15
33604 Bielefeld',
  'workStreet' => 'Hübschestr. 19',
  'workCity' => 'Köln',
  'workProvince' => 'Allgäu',
  'workPostalcode' => '33602',
  'workCountry' => 'Dänemark',
  'homePhone' => '+49 521 555123',
  'workPhone' => '+49 521 555456',
  'cellPhone' => '+49 177 555123',
  'fax' => '+49 521 555789',
  'pager' => '+49 123 555789',
  'birthday' => '1971-10-01',
  'title' => 'Senior Developer (äöü)',
  'role' => 'Developer (äöü)',
  'company' => 'Horde Project',
  'department' => 'äöü',
  'notes' => 'A German guy (äöü)',
  'website' => 'http://janschneider.de',
  'timezone' => 'Europe/Berlin',
  'latitude' => '52.516276',
  'longitude' => '13.377778',
  'photo' => file_get_contents(dirname(__FILE__) . '/az.png'),
  'phototype' => 'image/png',
);

$driver = new Turba_Driver(array());
$object = new Turba_Object($driver, $attributes);
$vcard = $driver->tovCard($object, '2.1');
echo $vcard->exportvCalendar() . "\n";
$vcard = $driver->tovCard($object, '3.0');
echo $vcard->exportvCalendar();

?>
