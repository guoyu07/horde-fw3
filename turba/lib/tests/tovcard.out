BEGIN:VCARD
VERSION:2.1
FN;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:=
Jan Schneider=F6
EMAIL:jan@horde.org
NICKNAME:yunosh
LABEL;HOME;ENCODING=QUOTED-PRINTABLE;CHARSET=ISO-8859-1:=
Sch=F6nestr. 15=0D=0A33604 Bielefeld
TEL;HOME:+49 521 555123
TEL;WORK:+49 521 555456
TEL;CELL:+49 177 555123
TEL;FAX:+49 521 555789
TEL;PAGER:+49 123 555789
BDAY:1971-10-01
TITLE;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:=
Senior Developer (=E4=F6=FC)
ROLE;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:=
Developer (=E4=F6=FC)
NOTE;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:=
A German guy (=E4=F6=FC)
URL:http://janschneider.de
TZ;VALUE=text:Europe/Berlin
GEO:52.516276,13.377778
PHOTO;ENCODING=b;TYPE=image/png:iVBORw0KGgoAAAANSUhEUgAAAAkA
 AAAJAgMAAACd/+6DAAAACVBMVEW6ABZmZmYAAACMtcxCAAAAAXRSTlMAQObY
 ZgAAABpJREFUCFtjYACBBgYmRgEIZmGBYAFGMAYBABVmAOEH9qP8AAAAAElF
 TkSuQmCC
N;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:=
Schneider=F6;Jan;K.;Mr.;
ORG;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:=
Horde Project;=E4=F6=FC
ADR;HOME;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:=
;;Sch=F6nestr. 15=0D=0A33604 Bielefeld;;;;
ADR;WORK;CHARSET=ISO-8859-1;ENCODING=QUOTED-PRINTABLE:=
;;H=FCbschestr. 19;K=F6ln;Allg=E4u;;D=E4nemark
BODY:
END:VCARD

BEGIN:VCARD
VERSION:3.0
FN:Jan Schneiderö
EMAIL:jan@horde.org
NICKNAME:yunosh
LABEL;TYPE=HOME:Schönestr. 15\n33604 Bielefeld
TEL;TYPE=HOME:+49 521 555123
TEL;TYPE=WORK:+49 521 555456
TEL;TYPE=CELL:+49 177 555123
TEL;TYPE=FAX:+49 521 555789
TEL;TYPE=PAGER:+49 123 555789
BDAY:1971-10-01
TITLE:Senior Developer (äöü)
ROLE:Developer (äöü)
NOTE:A German guy (äöü)
URL:http://janschneider.de
TZ;VALUE=text:Europe/Berlin
GEO:52.516276,13.377778
PHOTO;ENCODING=b;TYPE=image/png:wolQTkcNChoKAAAADUlIRFIAAAAJ
 AAAACQIDAAAAwp3Dv8OuwoMAAAAJUExURcK6ABZmZmYAAADCjMK1w4xCAAAA
 AXRSTlMAQMOmw5hmAAAAGklEQVQIW2NgAMKBBgYmRgEIZmHCgWABRjAGAQAV
 ZgDDoQfDtsKjw7wAAAAASUVORMKuQmDCgg==
N:Schneiderö;Jan;K.;Mr.;
ORG:Horde Project;äöü
ADR;TYPE=HOME:;;Schönestr. 15\n33604 Bielefeld;;;;
ADR;TYPE=WORK:;;Hübschestr. 19;Köln;Allgäu;;Dänemark
BODY:
END:VCARD